extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -926,13 +937,37 @@ public static Table readFlatPartitionedTable(
      * @param file the parquet file
      * @param readInstructions the instructions for customizations while reading
      * @return the table
-     * @see ParquetTableLocationKey#ParquetTableLocationKey(File, int, Map)
+     * @see ParquetTableLocationKey#ParquetTableLocationKey(File, int, Map, ParquetInstructions)
      * @see #readSingleFileTable(ParquetTableLocationKey, ParquetInstructions, TableDefinition)
      */
     public static Table readSingleFileTable(
             @NotNull final File file,
             @NotNull final ParquetInstructions readInstructions) {
-        final ParquetSingleFileLayout keyFinder = new ParquetSingleFileLayout(file);
+        return readSingleFileTable(file.toURI(), readInstructions);
+    }
+
+    /**
+     * Creates a single table via the parquet {@code filePath} using the provided {@code tableDefinition}.
+     * <p>
+     * Callers wishing to be more explicit (for example, to skip some columns) may prefer to call
+     * {@link #readSingleFileTable(File, ParquetInstructions, TableDefinition)}.
+     *
+     * @param filePath the parquet file path
+     * @param readInstructions the instructions for customizations while reading
+     * @return the table
+     * @see ParquetTableLocationKey#ParquetTableLocationKey(File, int, Map, ParquetInstructions)
+     * @see #readSingleFileTable(ParquetTableLocationKey, ParquetInstructions, TableDefinition)
+     */
+    public static Table readSingleFileTable(
```

Comment:
I have added a few new APIs which take strings instead of the usual File based API, so that we can pass S3 URIs using the strings.
And both File based and String based APIs are implemented using private URI based methods.